src/electos/ballotmaker/ballots/ballot_layout.py

```diff
@@ -139,42 +138,45 @@ def build_ballot() -> str:
     )
     doc.addPageTemplates(three_column_template)
     doc.addPageTemplates(one_column_template)
-    # add a ballot contest to the second frame (colomn)
-    layout_1 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_1)
-    )
-    layout_2 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_2)
-    )
-    layout_3 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_3)
-    )
-    layout_4 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_4)
-    )
-    layout_5 = BallotMeasureLayout(
-        BallotMeasureData(spacetown_data.ballot_measure_1)
-    )
-    layout_6 = BallotMeasureLayout(
-        BallotMeasureData(spacetown_data.ballot_measure_2)
-    )
+
+    candidate_contests = []
+    ballot_measures = []
+    # get contests
+    for count, contest in enumerate(ballot_data.contests, start=1):
+        title = contest.title
+        con_type = contest.type
+        logging.info(f"Found contest: {title} - {con_type}")
+        if con_type == CANDIDATE:
+            candidate_contests.append(contest)
+        elif con_type == BALLOT_MEASURE:
+            ballot_measures.append(contest)
+        else:
+            raise ValueError(f"Unknown contest type: {con_type}")
+    logging.info(f"Total: {count} contests.")
+
```

Comment:
Confirming what I think you explained already: the reason not to use `ballot_data.candidate_contests` and ballot_data.ballot_measure_contests` is to retain the order of the contests encountered when logging?

Reply:
Thanks for bringing this to my attention. I simply assumed I had to go through the contests, didn't realize that you had already split them up!

Reply:
Also added a backlink to this thread in the code.